### ISSUE TITLE
investigate a bit more on what labels mean in clustering.qmd

### ISSUE THREAD

**Issue by neelsoumya** [cambiotraining/ml-unsupervised#8]
* investigate a bit more on what labels mean in clustering.qmd
* ```{python ch4-evaluation}
#| warning: false

from sklearn.metrics import silhouette_score, adjusted_rand_score # Import evaluation metrics

# Hierarchical Clustering
agg = AgglomerativeClustering(linkage='average', metric='manhattan')
cluster_labels = agg.fit_predict(X)

# 1. Internal Evaluation: Silhouette Score
# Measures how well-separated clusters are based on the data itself.
silhouette = silhouette_score(X, cluster_labels, metric='manhattan')
print("Silhouette score")
print(silhouette)
print("Score is from -1 to 1. Higher is better")

# 2. External Evaluation: Adjusted Rand Index
# Compares our cluster labels to the true cancer type labels.
ari = adjusted_rand_score(labs, cluster_labels)
print("Adjusted Rand Index")
print(ari)
print("Compares to true labels. Score is from -1 to 1. Higher is better")

```

* Plain old visual evaluation

 - compare to labels of what these cell lines are
 
# TODO: XX what do these labels mean?

```{python ch4-labs-visual-evaluation}
#| warning: false

dendrogram(Z, labels=labs)

print(labs)

plt.figure()
plt.title('Hierarchical Clustering Dendrogram (NCI60)')
plt.xlabel('Cell Line (True Labels)')
plt.ylabel('Distance')
plt.xticks(rotation=90)
plt.tight_layout()
plt.show()

```
